source/agora/node/Ledger.d

```diff
@@ -356,6 +361,74 @@ public class Ledger
         return true;
     }
 
+    /***************************************************************************
+
+        Returns the public keys of the validators that signed the block at
+        the specified height, and and also not classified as 'missing' validator
+
+        Params:
+            height = block height
+
+        Returns: the public keys of the validators that signed the block at
+                 the specified height, and and also not
+                 classified as 'missing' validator
+
+    ***************************************************************************/
+
+    private Point[] getPKeysOfSignedNonMissingValidators (Height height) nothrow @safe
+    {
+        Point[] keys;
+        Block block;
+        try
+        {
+            block = this.storage.readBlock(height);
+        }
+        catch (Exception e) {
+            log.error("unable to read block at height: {}", height);
+            return null;
+        }
```

Comment:
This is not handled at the call site. In general, we should probably just let it `throw`, much better to do so than to propose invalid consensus data.

Reply:
This function is very deep in the call hierarchy, and its parents are all nothrow. I can change it, but that will remove nothrow from core functions like valueExternalized, getValidTXSet.... Should I do that?

One thing is that I should have checked for the return value in the calling site, which was missing.